ssl/handshake_server.cc

```diff
@@ -303,10 +303,23 @@ static void ssl_get_compatible_server_ciphers(SSL_HANDSHAKE *hs,
   uint32_t mask_a = 0;
 
   if (ssl_has_certificate(hs)) {
```

Comment:
Should we add a null check for hs here?  There's a lot of functions that just assume hs is safe to use,  tls1_call_ocsp_stapling_callback included.

Reply:
`SSL_HANDSHAKE` is essentially the structure that contains the state of the entire handshake. It's the center piece of the all the handshake state functions.
I assume that these aren't checked since `hs` already has to be non-null in order to progress to this stage. The internal `ssl_run_handshake` function is what runs the handshake code. Before the only place where it's called (`SSL_do_handshake`), we perform a NULL check on `hs` in `SSL_in_init`.
* https://github.com/aws/aws-lc/blob/b99e3d60b9be4e9ee07e0cefdc4c87ca8fbd1e9b/ssl/ssl_lib.cc#L846-L862
* https://github.com/aws/aws-lc/blob/b99e3d60b9be4e9ee07e0cefdc4c87ca8fbd1e9b/ssl/internal.h#L2181-L2184